cardano-cli/src/Cardano/CLI/Compatible/Transaction.hs

```diff
@@ -279,69 +277,29 @@ runCompatibleTransactionCmd
         )
         sbe
 
-    let certsRefInputs =
-          [ refInput
-          | (_, Just sWit) <- certsAndMaybeScriptWits
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-        votesRefInputs =
-          [ refInput
-          | VotingProcedures _ (TxVotingProcedures _ (BuildTxWith voteMap)) <- [votes]
-          , sWit <- Map.elems voteMap
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-        proposalsRefInputs =
-          [ refInput
-          | ProposalProcedures _ (TxProposalProcedures _ (BuildTxWith proposalMap)) <- [protocolUpdates]
-          , sWit <- Map.elems proposalMap
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-    validatedRefInputs <-
-      liftEither . first CompatibleTxCmdError . validateTxInsReference $
-        certsRefInputs <> votesRefInputs <> proposalsRefInputs
     let txCerts = mkTxCertificates sbe certsAndMaybeScriptWits
 
-    -- this body is only for witnesses
-    apiTxBody <-
-      firstExceptT CompatibleTxBodyError $
-        hoistEither $
-          createTransactionBody sbe $
-            defaultTxBodyContent sbe
-              & setTxIns (map (,BuildTxWith (KeyWitness KeyWitnessForSpending)) ins)
-              & setTxOuts allOuts
-              & setTxFee (TxFeeExplicit sbe fee)
-              & setTxCertificates txCerts
-              & setTxInsReference validatedRefInputs
+    transaction@(ShelleyTx _ ledgerTx) <-
+      firstExceptT CompatiblePParamsConversionError . hoistEither $
+        createCompatibleTx sbe ins allOuts fee protocolUpdates votes txCerts
+
+    let txBody = ledgerTx ^. A.txToTxBodyL sbe
```

Comment:
We need to directly use the ledger tx body type instead.

Reply:
https://github.com/IntersectMBO/cardano-cli/pull/1057/files#r1959731654